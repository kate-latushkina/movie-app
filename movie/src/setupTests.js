// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Enzyme, { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.render = render;
global.mount = mount;
global.toJson = toJson;

// Fail tests on any warning
console.error = message => {
  throw new Error(message);
};
