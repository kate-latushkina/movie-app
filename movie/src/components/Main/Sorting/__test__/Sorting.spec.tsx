import React from "react";
import renderer from "react-test-renderer";
import SortingSection from "../Sorting";
import "../../../../setupTests";

describe("Sorting component", () => {
  it("should render SortingSection snapshot", () => {
    const wrapper = renderer.create(<SortingSection />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
