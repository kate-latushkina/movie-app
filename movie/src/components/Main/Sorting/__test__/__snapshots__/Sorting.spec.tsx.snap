// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting component should render SortingSection snapshot 1`] = `
<div
  className="MuiGrid-root sc-dlfnbm jTNYpR"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root sc-bdfBwQ gfyZvD MuiInputLabel-animated"
    id="select-label"
  >
    Sort by
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-1"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Release date
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value="Release date"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
